File: app/opensearch/keyword_search.py
from opensearch.query_builder import QueryRequest, normalize_request


def build_keyword_query(req: QueryRequest) -> dict:
    req = normalize_request(req)
    return {
        "size": req.size,
        "query": {
            "multi_match": {
                "query": req.keyword,
                "fields": ["name^3", "brand^2", "description", "category"],
                "type": "best_fields",
            }
        },
    }

File: backend/opensearch/keyword_search.go
package opensearch

func BuildKeywordQuery(req QueryRequest) map[string]any {
	req = NormalizeRequest(req)
	return map[string]any{
		"size": req.Size,
		"query": map[string]any{
			"multi_match": map[string]any{
				"query":  req.Keyword,
				"fields": []string{"name^3", "brand^2", "description", "category"},
				"type":   "best_fields",
			},
		},
	}
}
